### PR TITLE
X763 original sample registration

### PR DESCRIPTION
**Pull request by seenanair**
Original sample registration

### CODE REVIEW

File: src/lib/validation/registrationValidation.ts [hunk 1]
```diff
@@ -94,12 +111,21 @@ export default class RegistrationValidation {
...
-    return validation.requiredString({
-      label: "Replicate Number",
-      restrictChars: /^[1-9]\d*[a-z]?$/,
-      errorMessage:
-        "Replicate Number must be a positive integer, optionally followed by a lower case letter.",
-    });
+    if (this.tissueSampleRegistration) {
+      return validation.optionalString({
+        label: "Replicate Number",
+        restrictChars: /^[1-9]\d*[a-z]?$/,
```
[the remark this answers is not in the document]
Comment (seenanair) [reply src/lib/validation/registrationValidation.ts:117]: /^[1-9][0-9]*[a-z]?$/ Is this a better validation?

[the remark this answers is not in the document]
Comment (khelwood) [reply src/lib/validation/registrationValidation.ts:117]: I think it's my mistake. I misread the pattern you were using. Sorry.

[the remark this answers is not in the document]
Comment (seenanair) [reply src/lib/validation/registrationValidation.ts:117]: That's ok 👍

File: src/mocks/handlers/solutionSampleHandlers.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    );
+    if (solutionSample) {
+      solutionSample.enabled = req.variables.enabled;
+      solutionSampleRepository.save(solutionSample);
+      return res(ctx.data({ setSolutionSampleEnabled: solutionSample }));
+    } else {
+      return res(
+        ctx.errors([
+          {
+            message: `Could not find equipment: "${req.variables.name}"`,
```
[the remark this answers is not in the document]
Comment (seenanair) [reply src/mocks/handlers/solutionSampleHandlers.ts]: Done

**Review by khelwood — APPROVED** [review]

**Pull request merged**